internal/ingress/controller/store/store.go

```diff
@@ -766,3 +827,54 @@ func (s k8sStore) Run(stopCh chan struct{}) {
 		go wait.Until(s.checkSSLChainIssues, 60*time.Second, stopCh)
 	}
 }
+
+func (s k8sStore) getCurrentPod() *corev1.Pod {
+
+	ns := os.Getenv("POD_NAMESPACE")
+	name := os.Getenv("POD_NAME")
+
+	for _, i := range s.listers.Pod.List() {
+		pod := i.(*corev1.Pod)
+
+		if pod.GetNamespace() == ns &&
+			pod.GetName() == name {
+			return pod
+		}
+	}
+
+	return nil
+}
+
+// ListControllerPods returns a list of ingress-nginx controller Pods
+func (s k8sStore) ListControllerPods() []*corev1.Pod {
+
+	curPod := s.getCurrentPod()
+
+	var pods []*corev1.Pod
+	for _, i := range s.listers.Pod.List() {
+		pod := i.(*corev1.Pod)
+
+		if !hasSameOwner(curPod, pod) {
+			continue
+		}
+
+		if pod.Status.Phase != corev1.PodRunning {
+			continue
+		}
+
+		pods = append(pods, pod)
+	}
+
+	return pods
+}
+
+func hasSameOwner(pod1, pod2 *corev1.Pod) bool {
```

Comment:
I can not think of any better option. One could use namespace but it would not be a complete solution since technically speaking there can be two ingress-nginx deployments under single namespace.